glib-macros/src/lib.rs

```diff
@@ -38,8 +40,48 @@ pub fn genum_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 /// [`glib::Value`]: value/struct.Value.html
 #[proc_macro_derive(GBoxed, attributes(gboxed))]
 #[proc_macro_error]
-pub fn gboxed_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+pub fn gboxed_derive(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
     let gen = gboxed_derive::impl_gboxed(&input);
     gen.into()
 }
+
+/// Attribute macro for defining flags using the `bitflags` crate.
+/// This macro will also define a `GFlags::get_type` function and
+/// the [`glib::Value`] traits.
+///
+/// The expected `GType` name has to be passed as macro attribute.
+/// The name and nick of each flag can also be optionally defined.
+/// Default name is the flag identifier in CamelCase and default nick
+/// is the identifier in kebab-case.
+/// Combined flags should not be registered with the `GType` system
+/// and so needs to be tagged with the `#[gflags(skip)]` attribute.
+///
+/// # Example
+///
+/// ```
+/// #[macro_use] extern crate glib;
+/// use glib::prelude::*;
+/// use glib::subclass::prelude::*;
+///
+/// #[gflags("MyFlags")]
+/// enum MyFlags {
+///     #[gflags(name = "Flag A", nick = "nick-a")]
+///     A = 0b00000001,
+///     #[gflags(name = "Flag B")]
+///     B = 0b00000010,
+///     #[gflags(skip)]
+///     AB = Self::A.bits() | Self::B.bits(),
+///     C = 0b00000100,
```

Comment:
Maybe we can add something to the macro so that it gives a compiler error if there's an unskipped overlapping value?

Reply:
Not sure how you'd check for overlapped values?

Reply:
Good question, that would require actually evaluating the code to get the bit pattern it evaluates to. So maybe not :)